### PR TITLE
[DOC] Merge gnmt.md and transformer.md into gnmt.md and removed trans…

### PR DESCRIPTION
**Pull request by StrayBird-ATSH**
…former.md (dmlc#1081)

## Description ##
Merge gnmt.md and transformer.md into gnmt.md and removed transformer.md. Update index.rst

### Essentials ###
- [☑] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [☑] Changes are complete (i.e. I finished coding on this PR)
- [☑] All changes have test coverage
- [☑] Code is well-documented

## Comments ##
- Merged two articles into one article

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1101/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1101/1/index.html

File: docs/examples/index.rst [hunk 1]
```diff
@@ -58,24 +58,18 @@ Machine Translation
 .. container:: cards
 
    .. card::
-      :title: Google Neural Machine Translation
+      :title: Google Neural Machine Translation with Transformer
```
Comment (leezu) [inline docs/examples/index.rst]: GNMT and Transformers are two different models so the proposed title doesn't make sense.

File: docs/examples/machine_translation/gnmt.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
-# Google Neural Machine Translation (GNMT)
+# Google Neural Machine Translation (GNMT) with Transformers
```
Comment (leezu) [inline docs/examples/machine_translation/gnmt.md]: You can rename this to "Machine Translation".
Please also change the filename from gnmt.md to translation.md in that case.

File: docs/examples/index.rst (hunk 1 above)
Comment (StrayBird-ATSH) [reply docs/examples/index.rst]: Thanks for your advice. I have made changes accordingly.

File: docs/examples/machine_translation/gnmt.md (hunk 2 above)
Comment (StrayBird-ATSH) [reply docs/examples/machine_translation/gnmt.md]: Thanks for your advice. I have made changes accordingly.

**Comment by mli** [comment]
Job PR-1101/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1101/3/index.html

**Comment by mli** [comment]
Job PR-1101/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1101/2/index.html

**Comment by leezu** [comment]
What's the motivation to have a single notebook that trains both models?
Do you plan to address the following points?
- Have a coherent notebook. For example, currently in the middle of the notebook we just re-import the same packages as already done at the beginning of the notebook http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1101/2/examples/machine_translation/machine_translation.html#Load-MXNet-and-GluonNLP 
- Avoid out-of-memory problems: The trained GNMT model is still taking up memory. This may cause problems when training the transformer model later on low-gpu-memory systems?

**Comment by eric-haibin-lin** [comment]
Shall we keep them in two separate notebooks but introduce a hierarchy?
- Tutorial
  - machine translation
    - GNMT
    - transformer

**Pull request closed without merging**